src/common/RelatedType.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Wrapper type for relation type definitions in entities.
+ * Used to circumvent ESM modules circular dependency issue caused by reflection metadata saving the type of the key.
+ *
+ * Usage example:
+ * @Entity()
+ * export default class User {
+ *
+ *     @OneToOne(() => Profile, profile => profile.user)
+ *     profile: Related<Profile>;
+ *
+ * }
+ */
+export type Related<T> = T;
```

Comment:
not sure I like this name. Maybe call it `Relation` ?

Reply:
Done